core/trino-main/src/main/java/io/trino/operator/output/PositionsAppender.java

```diff
@@ -11,17 +11,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package io.trino.operator.output;
 
 import io.trino.spi.block.Block;
-import io.trino.spi.block.BlockBuilder;
-import io.trino.spi.type.Type;
+import io.trino.spi.block.BlockBuilderStatus;
 import it.unimi.dsi.fastutil.ints.IntArrayList;
 
-import static java.util.Objects.requireNonNull;
+import javax.annotation.Nullable;
 
 public interface PositionsAppender
 {
-    void appendTo(IntArrayList positions, Block source, BlockBuilder target);
+    void appendTo(IntArrayList positions, Block source);
```

Comment:
Why not just `int[]` instead of an `IntArrayList`. Less dependencies

Reply:
the `IntArrayList .size()` is needed so this would need to be `int[], int` and for this IntArrayList seems cleaner to me.
I was thinking about using a different abstraction actually. `SelectedPositions` seems to fit the use case. Thoughts?